app/src/test/java/uk/ac/ebi/atlas/experiments/ExperimentDesignControllerWIT.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import uk.ac.ebi.atlas.configuration.TestConfig;
+
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@WebAppConfiguration
+@ContextConfiguration(classes = TestConfig.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ExperimentDesignControllerWIT {
+
+    private static final String URL = "/json/experiment-design/{experiment_accession}";
+
+    @Inject
+    private DataSource dataSource;
+
+    @Autowired
+    private WebApplicationContext wac;
+
+    private MockMvc mockMvc;
+
+    @BeforeAll
+    void populateDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(
+                new ClassPathResource("fixtures/experiment.sql"),
+                new ClassPathResource("fixtures/scxa_exp_design_column.sql"),
+                new ClassPathResource("fixtures/scxa_exp_design.sql"));
+        populator.execute(dataSource);
+    }
+
+    @AfterAll
+    void cleanDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(
+                new ClassPathResource("fixtures/scxa_exp_design-delete.sql"),
+                new ClassPathResource("fixtures/scxa_exp_design_column-delete.sql"),
+                new ClassPathResource("fixtures/experiment-delete.sql"));
+        populator.execute(dataSource);
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+    }
+
+    @Test
+    void hasExperimentDesignDataWithoutAnyPageNoSizeParameter() throws Exception {
+        mockMvc.perform(get(URL, "E-EHCA-2"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andExpect(jsonPath("$.headers").isArray())
+                .andExpect(jsonPath("$.headers").isNotEmpty())
+                .andExpect(jsonPath("$.data").isArray())
+                .andExpect(jsonPath("$.data").isNotEmpty())
+                .andExpect(jsonPath("$.headers").contains(List.of("contrastName", "referenceOrTest", "run")))
```

Comment:
I don't really like using magic strings here. Would be nice to introduce an `Enum` or something similar for these kind of values in the main code and use that in the test, too. I know the code was like this before, but unfortunately it is not really organised well. This is the case when we can use the `boy-scout` rule. Leave everything in a better way than you found it.

Reply:
If you mean the list of `"contrastName", "referenceOrTest", "run"`, they are only tested in this class, so I think it is fine to leave it like that. If we would like to improve/refactor the code somehow, we may raise another story regarding that.

Reply:
This test class has been removed already as its test subject was not needed anymore.